fsmonitor.c

```diff
@@ -83,7 +84,7 @@ void write_fsmonitor_extension(struct strbuf *sb, struct index_state *istate)
 	uint32_t ewah_size = 0;
 	int fixup = 0;
 
-	if (istate->fsmonitor_dirty->bit_size > istate->cache_nr)
+	if (!istate->split_index && istate->fsmonitor_dirty->bit_size > istate->cache_nr)
```

Comment:
1. Is a similar change needed in `tweak_fsmonitor`?  If so, I think that would point to a bigger issue as `ewah_each_bit(istate->fsmonitor_dirty, fsmonitor_ewah_callback, istate);` can `BUG` if called on a split_index.

2. This `BUG` (and the one in `read_fsmonitor_extension`) were introduced in git 2.24 in commit https://github.com/git/git/commit/3444ec2eb2be58c285d2bf04f39e6e9ea5eda9a2.  Should the commit with this fix be fast-tracked to the mailing list as its own patch?

cc: @dscho for his thoughts

Reply:
Hmm. I'll have to free up some brain cycles to think about this. So close to v2.24.0, I don't think we will get this into `master`, though. It only affects the split index feature, though, right? That's a fringe feature (and not a particularly well-designed one, I think), so I don't think that we will have to rush things with those fixes.

Reply:
> It only affects the split index feature, though, right?

Yep this issue is specific to split-index + fsmonitor, thanks!